pkg/utils/io/io.go

```diff
@@ -46,8 +46,8 @@ func NewCloser(close func() error) Closer {
 
 // Close is a convenience function to close a object that has a Close() method, ignoring any errors
 // Used to satisfy errcheck lint
-func Close(c Closer) {
+func Close(c Closer, log logr.Logger) {
```

Comment:
The `io.Close` is used extremely frequently in Argo CD. It would be really painful to add one more parameter. Instead can we global package level logger instance ?

Reply:
Thanks for the review! I've made the changes you have suggested, except for this one. May I suggest to copy this tiny function into Argo-CD instead of introducing a global logger just for this case? Would be really nice to avoid any globals in the library.

Reply:
Sounds good :) @ash2k can you please remove `Close` function as well as `Closer` interface with `inlineCloser` ? I'm working on moving them to Argo CD

Reply:
@alexmt Done! Looks better now in my opinion! Thanks.